testing/regress/ecl/parquet_types.ecl

```diff
@@ -0,0 +1,589 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+//class=parquet
+//Cover's data type's supported by ECL and arrow
+
+IMPORT Std;
+IMPORT Parquet;
+
+RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    BOOLEAN value;
+END;
+
+booleanDatasetOut := DATASET([
+    {000, 'aaa', TRUE},
+    {001, 'aab', FALSE}
+], RECORDDEF);
+
+ParquetIO.Write(booleanDatasetOut, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet', TRUE);
+
+booleanDatasetIn := ParquetIO.Read(RECORDDEF, '/var/lib/HPCCSystems/mydropzone/BooleanTest.parquet');
+
+joinResult := JOIN(booleanDatasetOut, booleanDatasetIn, LEFT.testid = RIGHT.testid AND LEFT.testname = RIGHT.testname AND LEFT.value = RIGHT.value, TRANSFORM(RECORDDEF, SELF := LEFT));
+
+booleanResult := IF(COUNT(booleanDatasetOut) = COUNT(booleanDatasetIn) AND COUNT(joinResult) = COUNT(booleanDatasetOut), 'Pass', 'Fail');
+
+INTEGER_RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    INTEGER value;
+END;
+
+integerDatasetOut := DATASET([
+    {-2147483648, 'min', -2147483648},
+    {2147483647, 'max', 2147483647}
+], INTEGER_RECORDDEF);
+
+ParquetIO.Write(integerDatasetOut, '/var/lib/HPCCSystems/mydropzone/IntegerTest.parquet', TRUE);
+
+integerDatasetIn := ParquetIO.Read(INTEGER_RECORDDEF, '/var/lib/HPCCSystems/mydropzone/IntegerTest.parquet');
+
+integerResult := IF(
+    COUNT(integerDatasetOut) = COUNT(integerDatasetIn) AND
+    COUNT(JOIN(integerDatasetOut, integerDatasetIn, LEFT.testid = RIGHT.testid AND LEFT.testname = RIGHT.testname AND LEFT.value = RIGHT.value)) = COUNT(integerDatasetOut),
+    'Pass', 'Fail: Integer data mismatch'
+);
+
+UNSIGNED_RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    UNSIGNED value;
+END;
+
+unsignedDatasetOut := DATASET([
+    {020, 'aan', 0},
+    {021, 'aao', 12345},
+    {022, 'aap', 4294967295}
+], UNSIGNED_RECORDDEF);
+
+ParquetIO.Write(unsignedDatasetOut, '/var/lib/HPCCSystems/mydropzone/UnsignedTest.parquet', TRUE);
+
+unsignedDatasetIn := ParquetIO.Read(UNSIGNED_RECORDDEF, '/var/lib/HPCCSystems/mydropzone/UnsignedTest.parquet');
+
+unsignedResult := IF(
+    COUNT(unsignedDatasetOut) = COUNT(unsignedDatasetIn) AND
+    COUNT(JOIN(unsignedDatasetOut, unsignedDatasetIn, LEFT.testid = RIGHT.testid AND LEFT.testname = RIGHT.testname AND LEFT.value = RIGHT.value)) = COUNT(unsignedDatasetOut),
+    'Pass', 'Fail: Unsigned data mismatch'
+);
+
+// Define schema for REAL type
+REAL_RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    REAL value;
+END;
+
+// Define schema for DECIMAL type
+DECIMAL_RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    DECIMAL10_2 value;
+END;
+
+// Define schema for STRING type
+STRING_RECORDDEF := RECORD
+    UNSIGNED testid;
+    STRING3 testname;
+    STRING value;
+END;
```

Comment:
Either move all the Record definitions to the same spot at the top of the file or move each of them to immediately before they are needed.